src/Microsoft.Extensions.Configuration.UserSecrets/PathHelper.cs

```diff
@@ -3,39 +3,60 @@
 
 using System;
 using System.IO;
+using Microsoft.Extensions.FileProviders;
+using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 
 namespace Microsoft.Extensions.Configuration.UserSecrets
 {
     public class PathHelper
     {
         internal const string Secrets_File_Name = "secrets.json";
+        internal const string Config_File_Name = "project.json";
 
-        public static string GetSecretsPath(string projectPath)
+        public static string GetSecretsPath(IFileProvider provider)
         {
-            if (projectPath == null)
+            if (provider == null)
             {
-                throw new ArgumentNullException(nameof(projectPath));
+                throw new ArgumentNullException(nameof(provider));
             }
 
-            var projectFilePath = Path.Combine(projectPath, "project.json");
-
-            if (!File.Exists(projectFilePath))
+            var fileInfo = provider.GetFileInfo(Config_File_Name);
+            if (fileInfo == null || fileInfo.Exists == false || fileInfo.PhysicalPath == null)
             {
                 throw new InvalidOperationException(
-                    string.Format(Resources.Error_Missing_Project_Json, projectFilePath));
+                    string.Format(Resources.Error_Missing_Project_Json, provider.GetFileInfo("/")?.PhysicalPath));
```

Comment:
Unable to locate project.json at '{0}' should be what if there isn't a physical path?